### ISSUE TITLE
TokenAndPositionEmbedding does not propagate keras masks

### ISSUE THREAD

**Issue by mattdangerw** [keras-team/keras-nlp#138]
`keras.layers.Embedding` with `mask_zero=True` will create an `output._keras_mask` as expected.

`keras_nlp.layers.TokenAndPositionEmbedding` with `mask_zero=True` will not.

https://colab.sandbox.google.com/gist/mattdangerw/f523842e3e7ab4f4124257ba1571e290/tokenandpositionembedding-mask-bug.ipynb
